docs/source/en/features/distributed_adafactor.md

```diff
@@ -0,0 +1,163 @@
+
+```python
+# Distributed Enviroment
+config = {}
+colossalai.launch(config=config, rank=rank, world_size=world_size,host="localhost", port=port, backend="nccl")
+
+# Parallism Group
+tp_size, zero_size = 2, 2
+use_zero = True if zero_size > 1 else False
+proc_mesh = ProcessGroupMesh(tp_size, zero_size)
+tp_group, dp_group = proc_mesh.get_group_along_axis(0), proc_mesh.get_group_along_axis(1)
+```
+
+### step 3. Initialize Module
+Build our model. We created an MLP using two Linear Layer.
+
+```python
+# Init a Tensor Paralleled Module
+class TPModel(nn.Module):
+    def __init__(self, linear1, linear2, tp_group=None):
+        super().__init__()
+        self.linear1 = Linear1D_Col.from_native_module(
+            linear1, process_group=tp_group, gather_output=False, overlap=True
+        )
+        self.linear2 = Linear1D_Row.from_native_module(linear2, process_group=tp_group, parallel_input=True)
+
+    def forward(self, x):
+        x = self.linear1(x)
+        x = self.linear2(x)
+        return x
+HEIGHT = 4096
+WIDTH = 4096
+tp_model = TPModel(copy.deepcopy(nn.Linear(HEIGHT, WIDTH)), copy.deepcopy(nn.Linear(HEIGHT, WIDTH)), tp_group).to(local_rank)
+
+# Get Module parameter
+tp_param_group = [p for n, p in tp_model.named_parameters()]
+```
+
+### step 4. Initialize Optimizer
+Then, We initialise the optimiser using the model parameter. Then, we set the distributed information for optimiser.
+
+```python
+# Init a Optimizer
+dist_optim = DistributedAdaFactor(tp_param_group)
+shard_to_param = {id(p):p for p in tp_param_group}
+
+# Setup distributed information for Optimizer
+dist_optim.setup_distributed(
+    tensor_parallel_group=tp_group,
+    data_parallel_group=dp_group,
+    shard_to_param=shard_to_param,
+    use_zero=use_zero,
+)
+```
+
+### step 5. Perform a forward and backward propagation for model and step the gradient
+
+```python
+# Random initialise dataset
+x = torch.randn(HEIGHT, WIDTH, device=local_rank)
+
+# Fwd and Bwd
+out_tp = tp_model(x)
+if zero_size > 1:
+    dist_optim.backward(out_tp.sum())
+else:
+    out_tp.sum().backward()
+
+# perform step for param and grad
+dist_optim.step()
+dist_optim.zero_grad()
+```
+
+## Run with booster
+We highly recommend users to use booster, a simple, easy to use, and efficient interface. The Code Below is the Distributed Adafactor launched with booster.
+
+```python
+# ==============================
+# Model Init
+# ==============================
+tp_model = TPModel()
+
+# ==============================
+# Optimizer Init
+# ==============================
+dist_optim = DistributedAdaFactor([p for n, p in tp_model.named_parameters()])
+
+# ==============================
+# Booster Init
+# ==============================
+plugin = TorchDDPPlugin()
+booster = Booster(plugin=plugin)
+criterion = lambda x: x.mean()
+tp_model, dist_optim, criterion, _, _ = booster.boost(tp_model, dist_optim, criterion)
+```
+
+## Performance
```

Comment:
performance is not necessary. Furthermore, we need a support matrix to indicate which plugin support which optimizer